src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -214,92 +386,193 @@
      * can be set with {@link #setSerialFilter(ObjectInputFilter) Config.setSerialFilter}.
      * Setting the {@code jdk.serialFilter} with {@link System#setProperty(String, String)
      * System.setProperty} <em>does not set the filter</em>.
-     * The syntax for each property is the same as for the
+     * The syntax for the property value is the same as for the
      * {@link #createFilter(String) createFilter} method.
-     *
+     * <p>
+     * If the Java virtual machine is started with the system property
+     * {@systemProperty jdk.serialFilterFactory}, its value names the class to configure the
+     * JVM-wide deserialization filter factory.
+     * If the system property is not defined, and the {@link java.security.Security} property
+     * {@code jdk.serialFilterFactory} is defined then it is used to configure the filter factory.
+     * The class must be public, must have a public zero-argument constructor, implement the
+     * {@link BinaryOperator {@literal BinaryOperator<ObjectInputFilter>}} interface, provide its implementation and
+     * be accessible via the {@linkplain ClassLoader#getSystemClassLoader() application class loader}.
+     * The filter factory configured using the system or security property during initialization
```

Comment:
What is the expected behaviour if the factory property is to set to a non-class or non-accessible class? The current implementation does (it probably should be more graceful) :

$ java -Djdk.serialFilterFactory=allow T
Exception in thread "main" java.lang.ExceptionInInitializerError
	at java.base/java.io.ObjectInputFilter$Config.<clinit>(ObjectInputFilter.java:537)
	at java.base/java.io.ObjectInputStream.<init>(ObjectInputStream.java:394)
	at T.main(T.java:5)
Caused by: java.lang.ClassNotFoundException: allow
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:466)
	at java.base/java.io.ObjectInputFilter$Config.<clinit>(ObjectInputFilter.java:519)
	... 2 more

Reply:
If the factory class can not be found, the exception must be fatal; 
continuing to run without the filter would be a security risk.
ExceptionInInitializerError was the closest I could find.
I'll improve the message;  Oddly, ExceptionInInitializer does not allow both a message and initCause().
And the stacktrace for the ClassNotFoundException is not going to be very interesting.